### PR TITLE
Replace attrs with dataclasses

### PR DESCRIPTION
**Pull request by benjeffery**
Now we are on 3.7, removes a dependancy.

### CODE REVIEW

File: python/requirements/CI-tests-pip/requirements.txt [hunk 1]
```diff
@@ -3,7 +3,6 @@ pytest==6.2.2
 pytest-xdist==2.2.1
 jsonschema==3.2.0
 h5py==3.1.0
```
Comment (jeromekelleher) [inline python/requirements/CI-tests-pip/requirements.txt:5]: Looks like there's still a h5py in here?

**Review by jeromekelleher — COMMENTED** [review]
Excellent! Looks like essentially a drop-in replacement.

**Review by jeromekelleher — APPROVED** [review]

File: python/requirements/CI-tests-pip/requirements.txt (hunk 1 above)
Comment (benjeffery) [reply python/requirements/CI-tests-pip/requirements.txt:5]: Yes, there needs to be one. Somehow we had two!!

Comment (jeromekelleher) [reply python/requirements/CI-tests-pip/requirements.txt:5]: Yep, my bad

**Pull request merged**